tools/gulptasks/test.js

```diff
@@ -415,23 +415,29 @@ Available arguments for 'gulp test':
 
             LogLib.message('Run `gulp test --help` for available options');
 
-            const KarmaServer = require('karma').Server;
+            const karma = require('karma');
+            const KarmaServer = karma.Server;
             const PluginError = require('plugin-error');
             const {
                 reporters: defaultReporters,
                 browserDisconnectTimeout: defaultTimeout
             } = require(KARMA_CONFIG_FILE);
 
             new KarmaServer(
-                {
-                    configFile: KARMA_CONFIG_FILE,
-                    reporters: argv.dots ? ['dots'] : defaultReporters,
-                    browserDisconnectTimeout: typeof argv.timeout === 'number' ? argv.timeout : defaultTimeout,
-                    singleRun: true,
-                    client: {
-                        cliArgs: argv
+                karma.config.parseConfig(
+                    KARMA_CONFIG_FILE,
+                    {
+                        reporters: argv.dots ? ['dots'] : defaultReporters,
+                        browserDisconnectTimeout: typeof argv.timeout === 'number' ? argv.timeout : defaultTimeout,
+                        singleRun: true,
+                        client: {
+                            cliArgs: argv
+                        }
+                    }, {
+                        promiseConfig: false,
+                        throwErrors: false
```

Comment:
Karma should throw errors. If it is not working without the flag, it should be fixed at the root.